packages/ember-metal/tests/tracked/validation_test.ts

```diff
@@ -85,32 +89,35 @@ if (EMBER_METAL_TRACKED_PROPERTIES) {
       }
 
       [`@test interaction with Ember object model (Ember computed property depending on tracked property)`](
-        assert
+        assert: Assert
       ) {
+        let self: EmberObject;
         class EmberObject {
-          constructor(name) {
+          name: Name;
+          constructor(name: Name) {
             this.name = name;
+            self = this;
           }
         }
 
         defineProperty(
           EmberObject.prototype,
           'full',
           computed('name', function() {
-            let name = get(this, 'name');
+            let name = get(self, 'name');
```

Comment:
I think you're having to introduce the `self` local variable because TypeScript can't infer the type of `this` inside this computed property. If that's the case, a better way to do this would be to annotate the `this` type:

```ts
computed('name', function(this: EmberObject) { /*...*/ })
```

Reply:
Cool! Done.